src/main/java/org/truffleruby/core/cast/ToRopeNode.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+
+package org.truffleruby.core.cast;
+
+import com.oracle.truffle.api.dsl.NodeChild;
+import com.oracle.truffle.api.dsl.Specialization;
+import org.truffleruby.core.rope.Rope;
+import org.truffleruby.core.string.ImmutableRubyString;
+import org.truffleruby.core.string.RubyString;
+import org.truffleruby.core.symbol.RubySymbol;
+import org.truffleruby.language.RubyContextSourceNode;
+import org.truffleruby.language.RubyNode;
+
+@NodeChild(value = "child", type = RubyNode.class)
+public abstract class ToRopeNode extends RubyContextSourceNode {
+
+    public abstract Rope executeToRope(Object object);
+
+    public static ToRopeNode create() {
+        return ToRopeNodeGen.create(null);
+    }
+
+    @Specialization
+    protected Rope coerceRubyString(RubyString string) {
+        return string.rope;
+    }
+
+    @Specialization
+    protected Rope coerceImmutableRubyString(ImmutableRubyString string) {
+        return string.rope;
+    }
+
+    @Specialization
+    protected Rope coerceSymbol(RubySymbol symbol) {
```

Comment:
Does Symbol needs to be handled hee?

Reply:
No. It's just something that has a `Rope` and made a bit of sense to me as a "cast" node.

Reply:
I think we should only add it if we need it.

We should also have a proper TypeError if the `object` is not matching any specialization, and not rely on the UnsupportedSpecializationException as that's thrown behind a transfer & deopt. In general, cast/coerce should handle any `Object` (either convert or raise) for correctness.

Reply:
What sort of exception do you want to see then? It's not something a user should ever see. The only use at the moment is to help out `@CreateCast` inside a primitive node. I'd prefer not to use coercion. I think that'll just make incorrect uses harder for a contributor to see. E.g., if I remove the `Symbol` specialization because we don't want to handle unknown cases yet, it would be able to coerce into a `String` and get caught by the fallback case.

Reply:
Looking at other cast nodes, `ToEncodingNode` returns `null` for its fallback case. I think that's the closest case to what I'm trying to do, which is an internal cast to extract a Java object from within a Ruby object. It looks a bit weird to me, but it would lead to an NPE and that should be easy enough to understand.

Reply:
I was thinking something like `IntegerCastNode` does with `typeErrorIsNotA`.
Yeah there is no need to do coercion here, but I meant whether it's a cast or coerce node, they should all have proper exceptions if the value is somehow not what we expect.

We could instead have a shouldNotReachHere() to make it clear the argument must be guaranteed to be a String before, but I feel it's all too easy to misuse CreateCast and forgot about limitations of cast/coerce nodes.

Reply:
Thanks. I forgot about `shouldNotReachHere()`. That's indeed what I was looking for.